src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAX_STORAGE];
+        this.values = (V[]) new Object[MAX_STORAGE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
+                values[i] = value;
+                return;
+            }
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < keys.length; i++) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
+                return values[i];
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        for (K key : keys) {
+            if (key != null) {
+                return size;
+            }
+        }
+        return size;
```

Comment:
```suggestion
        return size;
```
Why would you need all this if this line would return the same result?)

Reply:
My bad, I thought I fixed it, rookie mistake :)